marketing/models.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2021 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Model definitions for marketing and followup email
+campaigns.
+"""
+from django.contrib.auth import get_user_model
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.template import Template
+from django.utils.functional import cached_property
+from django.utils.timezone import now
+
+
+class EmailTemplate(models.Model):
+    """
+    Model to store the Email templates
+    """
+    name = models.CharField(
+        unique=True,
+        max_length=100,
+        help_text="Name of the email template. It can be used as an identifier for emails. Must be unique."
+    )
+    subject = models.CharField(
+        max_length=256,
+        help_text="Template for the subject of email"
+    )
+    html_body = models.TextField(
+        help_text="Template for the html body of the email."
+    )
+    plaintext_body = models.TextField(
+        help_text="Template for the text body of the email."
+    )
+    is_active = models.BooleanField(
+        default=False,
+        help_text="If the email is active or not. Emails will be sent for active templates only."
+    )
+    send_after_days = models.IntegerField(
+        validators=[MinValueValidator(1)],
+        help_text="Number of days after instance activation the email will be sent"
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        """
+        String representation
+        """
+        return self.name
+
+    @cached_property
+    def subject_template(self):
+        """
+        Get Template instance for subject.
+        """
+        return Template(self.subject)
+
+    @cached_property
+    def html_body_template(self):
+        """
+        Get Template instance for html_body.
+        """
+        return Template(self.html_body)
+
+    @cached_property
+    def plaintext_body_template(self):
+        """
+        Get Template instance for plaintext_body.
+        """
+        return Template(self.plaintext_body)
+
```

Comment:
Minor safeguard to avoid setting up emails after the prune time.
```suggestion

    def save(self, *args, **kwargs):
        # When saving, check if `send_after_days` is lower than MARKETING_DELETE_FOLLOWUP_EMAILS_AFTER_DAYS
        # otherwise emails will be repeatedly sent since the cleanup job will clear those settings.
        if self.send_after_days > settings.MARKETING_DELETE_FOLLOWUP_EMAILS_AFTER_DAYS:
            raise Exception(
                 "Configuring follow up emails after %i is not allowed", 
                 settings.MARKETING_DELETE_FOLLOWUP_EMAILS_AFTER_DAYS
            )
        super(Model, self).save(*args, **kwargs)

```